### DISCUSSION TITLE
NextJS Deployment Issue: 404 Errors on Pages Despite Functional API

### DISCUSSION THREAD

**Discussion by alaratte** [vercel/community#5211] [Help]
Hello,

I deployed my NextJS app on Vercel and use an API on another server for communication and data retrieval.

I'm experiencing 404 errors on some pages, even though the API is functioning correctly. I've checked the API logs, and there don't seem to be any issues. If there were problems with the API, why doesn't it build the next time I visit the page?

Here's a log from Vercel:

Erreur de la requête sur l'url TypeError: fetch failed at Object.fetch (node:internal/deps/undici/undici:11576:11) at async Fetcher (/var/task/.next/server/chunks/771.js:22:21) at async getPOI (/var/task/.next/server/chunks/771.js:56:24) at async getStaticProps (/var/task/.next/server/pages/[ville]/[action].js:521:21) { cause: SocketError: other side closed at TLSSocket.onSocketEnd (node:internal/deps/undici/undici:9790:26) at TLSSocket.emit (node:events:529:35) at endReadableNT (node:internal/streams/readable:1368:12) at process.processTicksAndRejections (node:internal/process/task_queues:82:21) { code: 'UND_ERR_SOCKET', socket: { localAddress: '169.254.76.1', localPort: 52706, remoteAddress: '109.234.162.207', remotePort: 443, remoteFamily: 'IPv4', timeout: undefined, bytesWritten: 1191, bytesRead: 4717 } } }

This issue persists with both NextJS 13 and 14. I've attempted various solutions, but none have been successful. Any assistance would be greatly appreciated.

Thanks !

**Comment by amyegan** [comment]
Hi @alaratte. Do you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use to get a better view of the problem?
